src/server/async/socket_server.cc

```diff
@@ -48,10 +48,13 @@ SocketConnection::SocketConnection(stream_protocol::socket socket,
   if (plasma_bulk_store != nullptr) {
     plasma_bulk_store_ = plasma_bulk_store->shared_from_this();
   }
-  auto gpu_bulk_store = server_ptr->GetBulkStore<GPUObjectID>();
+  
+  #ifdef ENABLE_GPU
+  auto gpu_bulk_store = server_ptr->GetBulkStore<GPUBulkStore>();
```

Comment:
To support heterogeneous object (some blobs are in GPU and others are in CPU), I suggest implement GPU-related  interface in original BulkStore to share the `object_map`.